File: Telerik/Reporting Q1 2014/Examples/CSharp/ReportLibrary/ProductCatalog/ProductCatalog.cs
namespace Telerik.Reporting.Examples.CSharp
{
    using System;
    using System.ComponentModel;
    using System.Drawing;
    using System.Globalization;
    using Telerik.Reporting;

    /// <summary>
    /// Summary description for Report1.
    /// </summary>
    [Description("Product Catalog Report. (Multilingual)")]
    public partial class ProductCatalog : Report
    {
        public ProductCatalog()
        {
            /// <summary>
            /// Required for telerik Reporting designer support
            /// </summary>
            InitializeComponent();

            //
            // TODO: Add any constructor code after InitializeComponent call
            //

            this.ItemDataBinding += new EventHandler(ProductCatalog_ItemDataBinding);
        }

        void ProductCatalog_ItemDataBinding(object sender, EventArgs e)
        {
            string cultureID = (string)this.ReportParameters["CultureID"].Value;

            CultureInfo cultureInfo = null;
            if (string.Compare(cultureID, "zh-CHT", true, CultureInfo.InvariantCulture) == 0)
            {
                cultureInfo = new CultureInfo("zh-TW", false);
            }
            else
            {
                cultureInfo = new CultureInfo(cultureID);
            }

            this.Culture = cultureInfo;
        }
    }
}

File: Telerik/Reporting Q1 2014/Examples/VB/ReportLibrary/ProductCatalog/ProductCatalog.vb

Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Globalization
Imports Telerik.Reporting

''' <summary>
''' Summary description for Report1.
''' </summary>
<Description("Product Catalog Report. (Multilingual)")> _
Partial Public Class ProductCatalog
    Inherits Report
    Public Sub New()
        ''' <summary>
        ''' Required for telerik Reporting designer support
        ''' </summary>
        InitializeComponent()

        '
        ' TODO: Add any constructor code after InitializeComponent call
        '

        AddHandler Me.ItemDataBinding, AddressOf Me.ProductCatalog_ItemDataBinding
    End Sub

    Private Sub ProductCatalog_ItemDataBinding(sender As Object, e As EventArgs)
        Dim cultureID As String = DirectCast(Me.ReportParameters("CultureID").Value, String)

        Dim cultureInfo__1 As CultureInfo = Nothing
        If String.Compare(cultureID, "zh-CHT", True, CultureInfo.InvariantCulture) = 0 Then
            cultureInfo__1 = New CultureInfo("zh-TW", False)
        Else
            cultureInfo__1 = New CultureInfo(cultureID)
        End If

        Me.Culture = cultureInfo__1
    End Sub
End Class
